File: Tan Khoa/1920ucnn.py
from math import *

n = int(input())
arr = list(map(int, input().split()))

g = 0 # UCLN của mảng arr
# chạy từng phần tử trong arr
    # gán g = gcd(g, v) ( v là phần tử của arr)

# khởi tạo d = -1 # ước nhỏ nhất của g
# Tìm ước nhỏ nhất của g
# chạy i từ 2 đến g(+1)
    # nếu i là ước của g
        # đặt d = i
        # dừng lại bằng lệnh break để lấy giá trị nhỏ nhất

print(d)

File: Tan Khoa/1920ucnn.cpp
#include <iostream>
#include <cmath>
#include <algorithm>

using namespace std;

int main() {
    // Input
    int n;  cin >> n;
    int arr[n];
    for (int i = 0; i < n; ++i)
        cin >> arr[i];
    // Solve
    int g = 0; // UCLN của
    for (int i = 0; i < n; ++i)
        g = __gcd(g, arr[i]);
    int ans = -1; // ước nhỏ nhất của g
    for (int i = 2; i <= g; ++i)
        if (g % i == 0) {
            ans = i;
            break;
        }
    cout << ans << endl;
    return 0;
}
